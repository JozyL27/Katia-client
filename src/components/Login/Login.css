.Login__container {
  width: 70%;
  font-size: 1.6rem;
  text-align: center;
  padding: 1rem 1rem 2rem;
}

.Login__header {
  margin: 1.5rem 0;
}

.Login__submit-button-div {
  display: flex;
  flex-direction: column;
  justify-content: center;
}

.Login__submit-button {
  width: 7.5rem;
  margin: 20px auto;
  border-radius: 4px;
  background-color: rgb(0,204,0);
  color: white;
  border-color: rgb(0,204,0);
  transition-duration: 0.4s;
}

.Login__submit-button:hover {
  background-color: rgb(0,204,0);
}

.Login__signup-link {
  width: 7.5rem;
  border: none;
  background: none;
  font-weight: bolder;
  color: rgb(0,204,0);
  cursor: pointer;
  font-size: 1.8rem;
  transition-duration: 0.5ms;
  margin: 0 auto;
}

.Login__or {
  margin: .75rem 0;
}