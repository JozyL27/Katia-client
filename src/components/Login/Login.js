import React, { Component } from 'react'
import {Link} from 'react-router-dom'
import TokenService from '../../services/token-service'
import AuthApiService from '../../services/auth-api-service'
import './Login.css'

export default class Login extends Component {
  static defaultProps = {
    onLoginSuccess: () => {}
  }

  state = { error: null }

  handleSubmitJwtAuth = ev => {
    ev.preventDefault()
    this.setState({error: null})
    const { email, password } = ev.target

    AuthApiService.postLogin({
      email: email.value,
      password: password.value
    })
      .then(res => {
        password.value = ''
        TokenService.saveAuthToken(res.authToken)
        this.props.onLoginSuccess()
        email.value = ''
      })
      .catch(res => {
        this.setState({error: res.error})
      })
  }

  render() {
    return (
      <main id='lp_main-flex'>
        <Link to='/' className='lp_lp-link'>Project Manhattan</Link>
        <section className='Login__container'>
          <h2 className='Login__header'>Login</h2>
          <form
            className='Login__form'
            onSubmit={this.handleSubmitJwtAuth}
          >
            <div role='alert'>
              {this.state.error && <p className='Login__error'>Something went wrong, please try again</p>}
            </div>
            <div className='Login__email-div lp_input-div'>
              <label htmlFor='Login__email'>
                Email
              </label>
              <input
                required
                name='email'
                id='Login__email'
                className='lp_input'>
              </input>
            </div>
            <div className='Login__password-div lp_input-div'>
              <label htmlFor='Login__password'>
                Password
              </label>
              <input
                required
                name='password'
                type='password'
                id='Login__password'
                className='lp_input'>
              </input>
            </div>
            <div className='Login__submit-button-div'>
              <button className='Login__submit-button lp_button' type='submit'>
                Login
              </button>
              <p className='Login__or'>or</p>
              <Link className='Login__signup-link lp_button' to='/signup'>Sign Up</Link>
            </div>
          </form>
        </section>
      </main>
    )
  }
}

