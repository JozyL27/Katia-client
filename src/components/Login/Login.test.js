import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import { BrowserRouter } from 'react-router-dom'
import Login from './Login'

describe('<Login />', () => {
  // Smoke test
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(
      <BrowserRouter>
        <Login />
      </BrowserRouter>, div)
    ReactDOM.unmountComponentAtNode(div)
  })
  // Snapshot test
  it('renders the UI as expected', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Login />
        </BrowserRouter>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})