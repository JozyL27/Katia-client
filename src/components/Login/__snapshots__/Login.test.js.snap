// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> renders the UI as expected 1`] = `
<main
  id="lp_main-flex"
>
  <a
    className="lp_lp-link"
    href="/"
    onClick={[Function]}
  >
    Project Manhattan
  </a>
  <section
    className="Login__container"
  >
    <h2
      className="Login__header"
    >
      Login
    </h2>
    <form
      className="Login__form"
      onSubmit={[Function]}
    >
      <div
        role="alert"
      />
      <div
        className="Login__email-div lp_input-div"
      >
        <label
          htmlFor="Login__email"
        >
          Email
        </label>
        <input
          className="lp_input"
          id="Login__email"
          name="email"
          required={true}
        />
      </div>
      <div
        className="Login__password-div lp_input-div"
      >
        <label
          htmlFor="Login__password"
        >
          Password
        </label>
        <input
          className="lp_input"
          id="Login__password"
          name="password"
          required={true}
          type="password"
        />
      </div>
      <div
        className="Login__submit-button-div"
      >
        <button
          className="Login__submit-button lp_button"
          type="submit"
        >
          Login
        </button>
        <p
          className="Login__or"
        >
          or
        </p>
        <a
          className="Login__signup-link lp_button"
          href="/signup"
          onClick={[Function]}
        >
          Sign Up
        </a>
      </div>
    </form>
  </section>
</main>
`;
